src/mgmt/cli_mgmt.c

```diff
@@ -380,22 +380,29 @@ int dc_mgmt_net_cfg(const char *name)
 	}
 
 	ofi_interface = getenv("OFI_INTERFACE");
+	ofi_domain = getenv("OFI_DOMAIN");
 	if (!ofi_interface) {
 		rc = setenv("OFI_INTERFACE", info.interface, 1);
 		if (rc != 0)
 			D_GOTO(cleanup, rc = d_errno2der(errno));
-	} else {
-		D_INFO("Using client provided OFI_INTERFACE: %s\n",
-			ofi_interface);
-	}
 
-	ofi_domain = getenv("OFI_DOMAIN");
-	if (!ofi_domain) {
+		/*
+		 * If we use the agent as the source, client env shouldn't be allowed to override
+		 * the domain. Otherwise we could get a mismatch between interface and domain.
+		 */
+		if (ofi_domain)
+			D_WARN("Using agent for network config. Ignoring client provided "
```

Comment:
I agree that there should be a warning, but this is a little obscure. I think we should assume that this is likely to be hit by someone who cargo-culted something or accidentally left it enabled or otherwise isn't really all that familiar with how the variables are supposed to be used.

I think it would be clearer to say something like, "Ignoring OFI_DOMAIN because OFI_INTERFACE is not set; using automatic configuration instead." This should indicate to the user that they either need to set OFI_INTERFACE or unset OFI_DOMAIN in order to quiet the warning.

I also think we should avoid the use of "client provided" in these messages because it exposes implementation details that aren't really relevant to the user.

Reply:
That's a better message, I'll update it.